doc/RELEASE.md

```diff
@@ -0,0 +1,57 @@
+## Release
+
+The release process follows a streamlined approach, no release branches involved.
+New releases can be major, minor or patch based releases, but always incrementing digits
+regarding the latest release version.
+
+### New Major.Minor.Patch version
+
+1. Create a new minor release branch from the HEAD of main:
+```sh
+git checkout -b release-vX.Y.Z
+```
+2. Update version (prefixed with **"v"**):
+```sh
+make prepare-release VERSION=vX.Y.Z
+```
+3. Verify local changes:
+```sh
+make install
+bin/kuadrantctl version
+```
+The output should be the new version, for example :
+```
+kuadrantctl v0.3.0 (ff779a1-dirty)
+```
+4. Commit and push:
+```sh
+git add .
+git commit -m "prepare-release: release-vX.Y.Z"
+git push origin release-vX.Y.Z
+```
+5. Create git tag:
+```sh
+git tag -s -m vX.Y.Z vX.Y.Z
+git push origin vX.Y.Z
+```
+6. In Github, [create release](https://github.com/Kuadrant/kuadrantctl/releases/new).
+
+* Pick recently pushed git tag
+* Automatically generate release notes from previous released tag
+* Set as the latest release
```

Comment:
small thing, but I guess you could do the verify bits at #45, then set it as latest release?

Reply:
Those verify steps, should not be some integration tests automated? 

We could have a workflow triggered on tagging that run those tests. Then the documentation explains that before creating the GH release, we should check the result of those tests. 

how does it sound?